File: binding/src/libqparallelanimationgroup.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qparallelanimationgroup.html
pub const qparallelanimationgroup = struct {

    /// New constructs a new QParallelAnimationGroup object.
    pub fn New() QtC.QParallelAnimationGroup {
        return qtc.QParallelAnimationGroup_new();
    }


    /// New2 constructs a new QParallelAnimationGroup object.
    pub fn New2(parent: ?*anyopaque) QtC.QParallelAnimationGroup {
        return qtc.QParallelAnimationGroup_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QParallelAnimationGroup_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qparallelanimationgroup.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Duration(self: ?*anyopaque, ) i32 {
        return qtc.QParallelAnimationGroup_Duration(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnDuration(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QParallelAnimationGroup_OnDuration(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseDuration(self: ?*anyopaque, ) i32 {
        return qtc.QParallelAnimationGroup_QBaseDuration(@ptrCast(self));
    }

    pub fn Event(self: ?*anyopaque, event: ?*anyopaque) bool {
        return qtc.QParallelAnimationGroup_Event(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QParallelAnimationGroup_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, event: ?*anyopaque) bool {
        return qtc.QParallelAnimationGroup_QBaseEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn UpdateCurrentTime(self: ?*anyopaque, currentTime: i32) void {
        qtc.QParallelAnimationGroup_UpdateCurrentTime(@ptrCast(self), currentTime);
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateCurrentTime(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QParallelAnimationGroup_OnUpdateCurrentTime(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateCurrentTime(self: ?*anyopaque, currentTime: i32) void {
        qtc.QParallelAnimationGroup_QBaseUpdateCurrentTime(@ptrCast(self), currentTime);
    }

    pub fn UpdateState(self: ?*anyopaque, newState: i32, oldState: i32) void {
        qtc.QParallelAnimationGroup_UpdateState(@ptrCast(self), @intCast(newState), @intCast(oldState));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateState(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32, i32) callconv(.c) void) void {
        qtc.QParallelAnimationGroup_OnUpdateState(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateState(self: ?*anyopaque, newState: i32, oldState: i32) void {
        qtc.QParallelAnimationGroup_QBaseUpdateState(@ptrCast(self), @intCast(newState), @intCast(oldState));
    }

    pub fn UpdateDirection(self: ?*anyopaque, direction: i32) void {
        qtc.QParallelAnimationGroup_UpdateDirection(@ptrCast(self), @intCast(direction));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateDirection(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QParallelAnimationGroup_OnUpdateDirection(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateDirection(self: ?*anyopaque, direction: i32) void {
        qtc.QParallelAnimationGroup_QBaseUpdateDirection(@ptrCast(self), @intCast(direction));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QParallelAnimationGroup_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qparallelanimationgroup.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QParallelAnimationGroup_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qparallelanimationgroup.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QParallelAnimationGroup_Delete(@ptrCast(self));
    }
};

File: binding/src/libqparallelanimationgroup.cpp
#include <QAnimationGroup>
#include <QEvent>
#include <QObject>
#include <QParallelAnimationGroup>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qparallelanimationgroup.h>
#include "libqparallelanimationgroup.h"
#include "libqparallelanimationgroup.hxx"

QParallelAnimationGroup* QParallelAnimationGroup_new() {
	 return new VirtualQParallelAnimationGroup();
}

QParallelAnimationGroup* QParallelAnimationGroup_new2(QObject* parent) {
	 return new VirtualQParallelAnimationGroup(parent);
}

libqt_string QParallelAnimationGroup_Tr(const char* s) {
	QString _ret = QParallelAnimationGroup::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QParallelAnimationGroup_Duration(const QParallelAnimationGroup* self) {
	return self->duration();
}

libqt_string QParallelAnimationGroup_Tr2(const char* s, const char* c) {
	QString _ret = QParallelAnimationGroup::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QParallelAnimationGroup_Tr3(const char* s, const char* c, int n) {
	QString _ret = QParallelAnimationGroup::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QParallelAnimationGroup_QBaseDuration(const QParallelAnimationGroup* self) {
	auto* vqparallelanimationgroup = dynamic_cast<const VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_Duration_IsBase(true);
	return vqparallelanimationgroup->duration();
}
}

// Auxiliary method to allow providing re-implementation
void QParallelAnimationGroup_OnDuration(const QParallelAnimationGroup* self, intptr_t slot) {
	auto* vqparallelanimationgroup = dynamic_cast<const VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_Duration_Callback(reinterpret_cast<VirtualQParallelAnimationGroup::QParallelAnimationGroup_Duration_Callback>(slot));
}
}

// Derived class handler implementation
bool QParallelAnimationGroup_Event(QParallelAnimationGroup* self, QEvent* event) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
	return vqparallelanimationgroup->event(event);
	} else {
	return self->QParallelAnimationGroup::event(event);
}
}

// Base class handler implementation
bool QParallelAnimationGroup_QBaseEvent(QParallelAnimationGroup* self, QEvent* event) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_Event_IsBase(true);
	return vqparallelanimationgroup->event(event);
}
}

// Auxiliary method to allow providing re-implementation
void QParallelAnimationGroup_OnEvent(QParallelAnimationGroup* self, intptr_t slot) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_Event_Callback(reinterpret_cast<VirtualQParallelAnimationGroup::QParallelAnimationGroup_Event_Callback>(slot));
}
}

// Derived class handler implementation
void QParallelAnimationGroup_UpdateCurrentTime(QParallelAnimationGroup* self, int currentTime) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
	vqparallelanimationgroup->updateCurrentTime(currentTime);
	} else {
	self->QParallelAnimationGroup::updateCurrentTime(currentTime);
}
}

// Base class handler implementation
void QParallelAnimationGroup_QBaseUpdateCurrentTime(QParallelAnimationGroup* self, int currentTime) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateCurrentTime_IsBase(true);
	vqparallelanimationgroup->updateCurrentTime(currentTime);
}
}

// Auxiliary method to allow providing re-implementation
void QParallelAnimationGroup_OnUpdateCurrentTime(QParallelAnimationGroup* self, intptr_t slot) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateCurrentTime_Callback(reinterpret_cast<VirtualQParallelAnimationGroup::QParallelAnimationGroup_UpdateCurrentTime_Callback>(slot));
}
}

// Derived class handler implementation
void QParallelAnimationGroup_UpdateState(QParallelAnimationGroup* self, int newState, int oldState) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
	vqparallelanimationgroup->updateState(static_cast<QAbstractAnimation::State>(newState), static_cast<QAbstractAnimation::State>(oldState));
	} else {
	self->QParallelAnimationGroup::updateState(static_cast<QAbstractAnimation::State>(newState), static_cast<QAbstractAnimation::State>(oldState));
}
}

// Base class handler implementation
void QParallelAnimationGroup_QBaseUpdateState(QParallelAnimationGroup* self, int newState, int oldState) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateState_IsBase(true);
	vqparallelanimationgroup->updateState(static_cast<QAbstractAnimation::State>(newState), static_cast<QAbstractAnimation::State>(oldState));
}
}

// Auxiliary method to allow providing re-implementation
void QParallelAnimationGroup_OnUpdateState(QParallelAnimationGroup* self, intptr_t slot) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateState_Callback(reinterpret_cast<VirtualQParallelAnimationGroup::QParallelAnimationGroup_UpdateState_Callback>(slot));
}
}

// Derived class handler implementation
void QParallelAnimationGroup_UpdateDirection(QParallelAnimationGroup* self, int direction) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
	vqparallelanimationgroup->updateDirection(static_cast<QAbstractAnimation::Direction>(direction));
	} else {
	self->QParallelAnimationGroup::updateDirection(static_cast<QAbstractAnimation::Direction>(direction));
}
}

// Base class handler implementation
void QParallelAnimationGroup_QBaseUpdateDirection(QParallelAnimationGroup* self, int direction) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateDirection_IsBase(true);
	vqparallelanimationgroup->updateDirection(static_cast<QAbstractAnimation::Direction>(direction));
}
}

// Auxiliary method to allow providing re-implementation
void QParallelAnimationGroup_OnUpdateDirection(QParallelAnimationGroup* self, intptr_t slot) {
	auto* vqparallelanimationgroup = dynamic_cast<VirtualQParallelAnimationGroup*>(self);
	if (vqparallelanimationgroup && vqparallelanimationgroup->isVirtualQParallelAnimationGroup) {
vqparallelanimationgroup->setQParallelAnimationGroup_UpdateDirection_Callback(reinterpret_cast<VirtualQParallelAnimationGroup::QParallelAnimationGroup_UpdateDirection_Callback>(slot));
}
}

void QParallelAnimationGroup_Delete(QParallelAnimationGroup* self) {
    delete self;
}

